tests/ui/lint/mutable_transmutes/lint.stderr

```diff
@@ -0,0 +1,124 @@
+error: transmuting &T to &mut T is undefined behavior, even if the reference is unused, consider instead using an UnsafeCell
+  --> $DIR/lint.rs:6:37
+   |
+LL |     let _a: [&mut u8; 2] = unsafe { transmute([&1u8; 2]) };
+   |                                     ^^^^^^^^^
+   |
+   = note: transmute from `[&u8; 2][0]` to `[&mut u8; 2][0]`
+   = note: `#[deny(mutable_transmutes)]` on by default
+
+error: converting &T to &UnsafeCell<T> is error-prone, rarely intentional and may cause undefined behavior
+  --> $DIR/lint.rs:45:29
+   |
+LL |         let _: A = unsafe { transmute(&1u8) };
+   |                             ^^^^^^^^^
+   |
+   = note: conversion from `*&u8` to `(*main::A.a.b.c).d`
+   = note: `#[deny(unsafe_cell_transmutes)]` on by default
+
+error: converting &T to &UnsafeCell<T> is error-prone, rarely intentional and may cause undefined behavior
+  --> $DIR/lint.rs:47:29
+   |
+LL |         let _: A = unsafe { transmute(E { e: F { f: &G { g: H { h: 0 } } } }) };
+   |                             ^^^^^^^^^
+   |
+   = note: conversion from `(*main::E.e.f).g.h` to `(*main::A.a.b.c).d`
+
+error: converting &T to &UnsafeCell<T> is error-prone, rarely intentional and may cause undefined behavior
+  --> $DIR/lint.rs:50:22
+   |
+LL |             unsafe { transmute(E { e: F { f: &G { g: H { h: 0 } } } }) };
+   |                      ^^^^^^^^^
+   |
+   = note: conversion from `(*main::E.e.f).g.h` to `*&UnsafeCell<u8>`
+
+error: transmuting &T to &mut T is undefined behavior, even if the reference is unused, consider instead using an UnsafeCell
+  --> $DIR/lint.rs:55:32
+   |
+LL |     let _a: &mut u8 = unsafe { transmute(&1u8) };
+   |                                ^^^^^^^^^
+   |
+   = note: transmute from `&u8` to `&mut u8`
+
+error: converting &T to &UnsafeCell<T> is error-prone, rarely intentional and may cause undefined behavior
+  --> $DIR/lint.rs:59:40
+   |
+LL |     let _a: &UnsafeCell<u8> = unsafe { transmute(&1u8) };
+   |                                        ^^^^^^^^^
+   |
+   = note: conversion from `*&u8` to `*&UnsafeCell<u8>`
+
+error: transmuting &T to &mut T is undefined behavior, even if the reference is unused, consider instead using an UnsafeCell
+  --> $DIR/lint.rs:77:48
+   |
+LL |         let _: Foo<&'static mut u8> = unsafe { transmute(Other(&1u8, &1u8)) };
+   |                                                ^^^^^^^^^
+   |
+   = note: transmute from `main::Other.1` to `main::Foo<&mut u8>.b.0.0`
+
+error: converting &T to &UnsafeCell<T> is error-prone, rarely intentional and may cause undefined behavior
+  --> $DIR/lint.rs:79:56
+   |
+LL |         let _: Foo<&'static UnsafeCell<u8>> = unsafe { transmute(Other(&1u8, &1u8)) };
+   |                                                        ^^^^^^^^^
+   |
+   = note: conversion from `*main::Other.1` to `*main::Foo<&UnsafeCell<u8>>.b.0.0`
```

Comment:
It's a bit hard to untangle the formatting machinery and I feel like this could be clearer. Here's a diagnostic I'd like to see instead:


```suggestion
note: this field... (`1` in `main::Other`)
  --> file.rs:LL:CC
   |
LL | struct Other(&'static u8, &'static u8);
   |                           ^^^^^^^^^^^
   |
   = note: this field has type `&u8`
   |
note: ...is being converted to this (`b.0.0` in `main::Foo<&UnsafeCell<u8>>`)
  --> file.rs:LL:CC
   |
LL | struct Baz<T>(T);
   |               ^
   |
   = note: this field has type `&UnsafeCell<u8>`
```
 
Storing the relevant information for this (instead of what is done currently) would probably reduce the complexity in the code.